foursight_core/react/api/aws_network.py

```diff
@@ -0,0 +1,421 @@
+import boto3
+import copy
+import json
+import re
+from typing import Callable, Optional, Union
+from dcicutils.obfuscation_utils import obfuscate_dict
+from .misc_utils import memoize, sort_dictionary_by_case_insensitive_keys
+
+
+def _filter_boto_description_list(description: dict,
+                                  name: str,
+                                  predicate: Optional[Union[str, re.Pattern, Callable]] = None,
+                                  create_record: Optional[Callable] = None) -> list:
+    """
+    Returns the list of AWS items from the given boto3 "describe" object (e.g. describe_vpc,
+    describe_subnets, describe_security_groups) which have tags names matching the given tag
+    predicate. This predicate may be (1) a string in which case it is used to match (case-sensitive)
+    the tag name exactly, or as a prefix if it ends with an asterisk; or (2) a regular expression
+    against which the tag name will be matched, or (3) a function which will be called with
+    each tag name and which should return True iff the tag name should be considered a match.
+    If no matches then returns empty list. Raises exception on error.
+    """
+    results = []
+    if isinstance(predicate, str):
+        if predicate.endswith("*"):
+            predicate_filter = lambda tag: isinstance(tag, str) and tag.startswith(predicate[0:len(predicate) - 1])
```

Comment:
I think something like this (untested) is easier to read:

```suggestion
            predicate_name = predicate[:-1]
            predicate_filter = lambda tag: isinstance(tag, str) and tag.startswith(predicate_name)
```

Although if the reason for the `ininstance(tag, str)` test is to screen out `None`, and if the empty string is not in play, I would go with just testing for `tag` as a boolean, since if it can only be a `None` or a `str`, it's adequate to test for truth (_unless_, as I say, you expect that the empty string could occur).

```suggestion
            predicate_name = predicate[:-1]
            predicate_filter = lambda tag: tag and tag.startswith(predicate_name)
```

(If you're in doubt, it's till fine to use `isinstance` as in the first of the two rewrites above,  I just find that a kind of ugly test. )

Reply:
Got it, sounds good, made change, thanks.